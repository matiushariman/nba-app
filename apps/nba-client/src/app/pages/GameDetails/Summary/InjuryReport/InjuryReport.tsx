import InjuryList from './InjuryList';
import { Card } from '../../../../components/ui';

const InjuryReport = () => (
  <Card className="p-5 grid gap-3 divide-y">
    <div>
      <p className="text-2xl font-bold uppercase">Injury report</p>
    </div>
    <div className="pt-5">
      <InjuryList />
    </div>
  </Card>
);

export default InjuryReport;
