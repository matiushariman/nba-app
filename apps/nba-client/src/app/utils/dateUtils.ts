import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
