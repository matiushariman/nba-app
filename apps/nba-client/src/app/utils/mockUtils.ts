export const shouldUseMock = process.env['NODE_ENV'] === 'development';
