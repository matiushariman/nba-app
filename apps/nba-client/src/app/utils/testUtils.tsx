import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
