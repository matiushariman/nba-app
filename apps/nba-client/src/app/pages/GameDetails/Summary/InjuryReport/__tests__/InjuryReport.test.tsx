import InjuryReport from '..';
import { render } from '../../../../../utils/testUtils';

jest.mock('../../../../../hooks/useGetBoxscore', () => ({
  selectInjuredPlayers: jest.fn().mockReturnValueOnce([
    {
      name: 'Anthony Davis',
      personId: 1,
      teamId: 1,
      teamTricode: 'LAL',
      jerseyNum: 3,
    },
  ]),
}));

describe('pages/GameDetails/Summary/InjuryReport', () => {
  it('should render InjuryReport that matches snapshot', () => {
    const { baseElement } = render(<InjuryReport />);

    expect(baseElement).toMatchSnapshot();
  });
});
