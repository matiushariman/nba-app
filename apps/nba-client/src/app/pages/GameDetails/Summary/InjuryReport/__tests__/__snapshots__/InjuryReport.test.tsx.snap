// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/GameDetails/Summary/InjuryReport should render InjuryReport that matches snapshot 1`] = `
<body>
  <div>
    <div
      class="p-5 grid gap-3 divide-y shadow-md bg-white rounded"
    >
      <div>
        <p
          class="text-2xl font-bold uppercase"
        >
          Injury report
        </p>
      </div>
      <div
        class="pt-5"
      >
        <div
          class="divide-y"
        >
          <div
            class="pt-2 pb-2 flex items-center gap-2"
          >
            <div>
              <div
                class="w-[48px] h-[48px]"
              >
                <img
                  alt="Anthony Davis headshot"
                  class="mr-2 bg-gray-300 shadow-sm rounded-full w-full h-full object-cover"
                  loading="lazy"
                  src="https://cdn.nba.com/headshots/nba/latest/260x190/1.png"
                />
              </div>
            </div>
            <div>
              <img
                alt="LAL logo"
                height="24"
                loading="lazy"
                src="https://cdn.nba.com/logos/nba/1/global/L/logo.svg"
                width="24"
              />
            </div>
            <div
              class="grid"
            >
              <p
                class="text-sm"
              >
                Anthony Davis
              </p>
              <p
                class="text-xs text-gray-500"
              >
                LAL
                 | #
                3
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
